content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/RealContentScopeScripts.kt

```diff
@@ -0,0 +1,150 @@
+            updateJS = true
+        }
+
+        if (cachedUserUnprotectedDomains != allowList.userWhiteList) {
+            cacheUserUnprotectedDomains(allowList.userWhiteList)
+            updateJS = true
+        }
+
+        val userPreferencesJson = getUserPreferencesJson(pluginParameters.preferences)
+        if (cachedUserPreferencesJson != userPreferencesJson) {
+            cachedUserPreferencesJson = userPreferencesJson
+            updateJS = true
+        }
+
+        if (!this::cachedContentScopeJS.isInitialized || updateJS) {
+            cacheContentScopeJS()
+        }
+        return cachedContentScopeJS
+    }
+
+    private fun getPluginParameters(): PluginParameters {
+        var config = ""
+        var preferences = ""
+        val plugins = pluginPoint.getPlugins()
+        plugins.forEach { plugin ->
+            if (config.isNotEmpty()) {
+                config += ","
+            }
+            config += plugin.config()
+
+            plugin.preferences()?.let { pluginPreferences ->
+                if (preferences.isNotEmpty()) {
+                    preferences += ","
+                }
+                preferences += pluginPreferences
+            }
+        }
+        return PluginParameters(config, preferences)
+    }
+
+    private fun cacheUserUnprotectedDomains(userUnprotectedDomains: List<String>) {
+        cachedUserUnprotectedDomains.clear()
+        if (userUnprotectedDomains.isEmpty()) {
+            cachedUserUnprotectedDomainsJson = emptyJsonList
+        } else {
+            cachedUserUnprotectedDomainsJson = getUserUnprotectedDomainsJson(userUnprotectedDomains)
+            cachedUserUnprotectedDomains.addAll(userUnprotectedDomains)
+        }
+    }
+
+    private fun cacheContentScopeJS() {
+        val contentScopeJS = contentScopeJSReader.getContentScopeJS()
+
+        cachedContentScopeJS = contentScopeJS
+            .replace(contentScope, cachedContentScopeJson)
+            .replace(userUnprotectedDomains, cachedUserUnprotectedDomainsJson)
+            .replace(userPreferences, cachedUserPreferencesJson)
+    }
+
+    private fun getUserUnprotectedDomainsJson(userUnprotectedDomains: List<String>): String {
+        val type = Types.newParameterizedType(MutableList::class.java, String::class.java)
+        val moshi = Builder().build()
+        val jsonAdapter: JsonAdapter<List<String>> = moshi.adapter(type)
+        return jsonAdapter.toJson(userUnprotectedDomains)
+    }
+
+    private fun getUserPreferencesJson(userPreferences: String): String {
+        if (userPreferences.isEmpty()) {
+            return "{${getVersionNumberKVP()}}"
+        }
+        return "{$userPreferences,${getVersionNumberKVP()}}"
+    }
+
+    private fun getVersionNumberKVP() = "\"versionNumber\":${appBuildConfig.versionCode}"
+
+    private fun getContentScopeJson(config: String): String = (
+        "{\"features\":{$config},\"unprotectedTemporary\":$emptyJsonList}"
```

Comment:
You are not passing the unprotectedTemporary so for those domains the gpc dom signal will be sent (if then script can actually handle that). If the script doesn't handle it, then it's ok but if the scripts handles them we need to pass them here instead of an empty JSON.